src/corona_hakab_model/agent.py

```diff
@@ -80,18 +102,64 @@ def add_agent(self,agent_snapshot):
 
     def export(self):
         num_sick = len(self.agent_snapshots)
-        export_dict = {"agent indexes":[0]*num_sick,"geographic_circles":[0]*num_sick,"age":[0]*num_sick}
-        social_circles = {connection_type.name:[0]*num_sick for connection_type in ConnectionTypes}
+        export_dict = {"agent indexes":[""]*num_sick,"geographic_circles":[""]*num_sick,"age":[""]*num_sick}
+        social_circles = {connection_type.name:[""]*num_sick for connection_type in ConnectionTypes}
+        social_circles_ids = {connection_type.name + self.ID_SUFFIX:[""]*num_sick for connection_type in ConnectionTypes}
+        social_circles = {**social_circles,**social_circles_ids}
+
         for index, agent_snapshot in enumerate(self.agent_snapshots):
             export_dict["agent indexes"][index] = agent_snapshot.index
             export_dict["geographic_circles"][index] = agent_snapshot.geographic_circle
             export_dict["age"][index] = agent_snapshot.age
             for social_circle_snapshot in agent_snapshot.social_circles:
                 social_circles[social_circle_snapshot.type][index] = social_circle_snapshot.num_members
+                social_circles[social_circle_snapshot.type + self.ID_SUFFIX][index] = social_circle_snapshot.id
         export_dict = {**export_dict,**social_circles}
         df_export_sick = pd.DataFrame(export_dict)
         df_export_sick.to_csv(self.EXPORT_OUTPUT_DIR + self.EXPORT_FILE_NAME,index=False)
 
+
+class InitialAgentsConstraints:
+    AGE = 'age'
+    GEOGRAPHIC_CIRCLE = "geographic_circles"
+    RANGE_DELIMITER = '~'
+    def __init__(self,constraints_file_path=None):
+        self.constraints = self.parse_constraints(constraints_file_path)
+
+    def parse_constraints(self,constraints_file_path):
+        if constraints_file_path is None:
+            return None
+        df_constraints = pd.read_csv(constraints_file_path)
+        constraints = []
+        for index, row in df_constraints.iterrows():
+            constraints.append(self.parse_row(row))
+        return constraints
+
+    def parse_row(self,row):
+        min_age,max_age = self.parse_range(row[self.AGE])
+        geographic_circle = row[self.GEOGRAPHIC_CIRCLE]
+        social_circle_constraints = []
+        for connection_type in ConnectionTypes:
+            min_num,max_num = self.parse_range(row[connection_type.name])
+            social_circle_constraints.append(SocialCircleConstraint(min_num,max_num,connection_type))
+        return AgentConstraint(min_age,max_age,geographic_circle,social_circle_constraints)
+
+    def parse_range(self,range_element):
+        if type(range_element) is int or type(range_element) is float:
+            return range_element,range_element
+        assert type(range_element) is str
+        split_range = range_element.split(self.RANGE_DELIMITER)
+        assert 0< len(split_range) <= 2
```

Comment:
If you assert it is better that you print something meaningfull

Reply:
Fixed to exception with meaningful message